### PR TITLE
Bye bye submodules

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

DEMO: https://www.loom.com/share/c8e4955578df4e50886f67dd82d48c49

This gets rids of private submodules by default.

Worked on a custom `git-setup.sh` script that will allow us to opt-in if we want to work on any specific submodule. We will remove `.gitmodules` from the main repo to avoid public users to auto-initialize it.

This will allow enterprise users to run `git-setup.sh api` to check if they have access to the repo and if not link them to our console. And if they have access, initialize the `.gitmodules` with only the api module setup so they don't get any console, website , etc. errors.

For the core team we can run `git-setup.sh api console website`  to get all available submodules to work locally.

Additionally added a `preinstall` `make` command that will check for the presence of a `.gitmodules` file and run `git-setup.sh api console website` only if not present.

### Downsides

- `yarn.lock` will be different between free, enterprise and core team members. Default will be public.
- Requires custom scripts for deploying modules in Vercel (which we already do anyways)

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### As a public user

- [ ] Clone this branch
- [ ] Run `yarn`
- [ ] All should go normal and an empty `.gitmodules` should be created

### As an enterprise user with API access

- [ ] Clone this branch
- [ ] Run `yarn`
- [ ] All should go normal  you should have `apps/api` in `.gitmodules`

### As an core team memeber

- [x] Clone this branch
- [x] Run `yarn`
- [x] All should go normal  you should have api, website and console in `.gitmodules`

### CODE REVIEW

File: Makefile
Comment (zomars) [inline Makefile:4]: If there's no `.gitmodules` file it will run the setup on `make`. This prevents running this command more than necessary.

File: package.json
Comment (zomars) [inline package.json:53]: Before installing runs setup only if not has run once already.

File: git-setup.sh
Comment (zomars) [inline git-setup.sh:13]: This makes a call to check if we have access to that repo

File: git-setup.sh
Comment (zomars) [inline git-setup.sh:21]: We prompt to go to console only when getting api module

File: git-setup.sh
Comment (zomars) [inline git-setup.sh:15]: Create a `.gitmodules` file if it doesn't exists

File: git-setup.sh
Comment (zomars) [inline git-setup.sh:17]: Initialize only the modules we have access to.

**Review by zomars — COMMENTED** [review]
Self-review done

File: .gitignore
Comment (zomars) [inline .gitignore:92]:
```suggestion
```

File: git-setup.sh
Comment (hariombalhara) [inline git-setup.sh:2]: Possible to use https://github.com/google/zx instead of shell script? 
Makes it easy to understand and contribute in it.

File: Makefile
Comment (hariombalhara) [inline Makefile:1]: Maybe we can avoid the use of makefile by checking presence of  .gitmodules in the script itself?

File: git-setup.sh
Comment (hariombalhara) [inline git-setup.sh:25]: This would unstage any staged changed as well. Can we do it in a way that achieves what's needed without a sideffect

File: Makefile
Comment (hariombalhara) [inline Makefile:4]: It would add(try to add) all three submodules for non-core team members as well. Shouldn't we let the core team members run this command manually?

**Review by hariombalhara — COMMENTED** [review]
Left some comments

File: Makefile
Comment (zomars) [reply Makefile:1]: It has global support and I'd rather not reinvent the wheel. It's perfect for this exact use case

File: git-setup.sh
Comment (zomars) [reply git-setup.sh:25]: Not that I'm aware of. This will only run once tho. Suggestions always welcome.

File: Makefile
Comment (zomars) [reply Makefile:4]: It will fail silently without disrupting anything. Did you try this locally?

**Review by emrysal — CHANGES_REQUESTED** [review]
After checkout and running `yarn` I get the following error 🤔 

```
xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

File: Makefile
Comment (zomars) [reply Makefile]: Got rid of Makefile cc @emrysal

File: git-init.sh
Comment (emrysal) [inline git-init.sh:4]: NIT typo

File: git-setup.sh
Comment (emrysal) [inline git-setup.sh:24]: Also typo NIT

**Review by emrysal — APPROVED** [review]
Approved, but if it can happen that there is a git reset with staged changes I would like a prompt that my staged changes are about to be reset, future improvement.

**Comment by krumware** [comment]
FYI, the use of git@github.com in the repo paths requires users to have ssh keys setup properly in their shell for git authentication. Could be worth mentioning in the docs for folks who errantly receive `You don't have access to: 'api' module.`

**Comment by krumware** [comment]
unexpected operator errors on Ubuntu (ignore the no-access, that's intentional)

```
Setting up 'api' module...
./git-setup.sh: 28: [: api: unexpected operator
You don't have access to: 'api' module.
Setting up 'website' module...
You don't have access to: 'website' module.
Setting up 'console' module...
./git-setup.sh: 28: [: website: unexpected operator
./git-setup.sh: 28: [: console: unexpected operator
You don't have access to: 'console' module.
```

**Comment by emrysal** [comment]
@krumware Committed in a fix, should work now 👍

**Pull request merged**
